tools/nnyeah/nnyeah/ConstructorTransforms.cs

```diff
@@ -90,19 +109,87 @@ bool IsNSObjectDerived (TypeReference? typeReference)
 
 		public void ReworkAsNeeded (TypeDefinition definition)
 		{
+			// Rework code blocks _before_ fixing the constructors, so we can tell which are
+			// invocations to the "old" IntPtr variant
+			ReworkCodeBlockAsNeeded (definition.Properties.SelectMany (p => Reworker.PropMethods (p)));
+			ReworkCodeBlockAsNeeded (definition.Methods);
+			ReworkCodeBlockAsNeeded (definition.Events.SelectMany (e => Reworker.EventMethods (e)));
+
 			if (IsNSObjectDerived (definition.BaseType)) {
 				if (definition.GetConstructors ().FirstOrDefault (IsIntPtrCtor) is MethodDefinition ctor) {
 					// How many instructions it takes to invoke a 1 or 2 param base ctor
 					// We can not safely process things that might store off the IntPtr
 					// or other such insanity, so just fail fast
 					if (ctor.Body.Instructions.Count > 7) {
-						throw new ConversionException (Errors.E0016);
+						throw new ConversionException (Errors.E0016, definition);
 					}
 					ctor.Parameters [0].ParameterType = NewNativeHandleTypeDefinition;
 					Transformed?.Invoke (this, new TransformEventArgs (ctor.DeclaringType.FullName,
 						ctor.Name, "IntPtr", 0, 0));
 				}
 			}
+
+		}
+
+		void ReworkCodeBlockAsNeeded (IEnumerable<MethodDefinition> methods)
+		{
+			foreach (var method in methods) {
+				ReworkCodeBlockAsNeeded (method);
+			}
+		}
+
+		// (Instruction Index, Number of Ctor Arguments)
+		List<(int, int)> FindConstructorInstructionIndex (IList<Instruction> instructions)
+		{
+			var index = new List<(int, int)> ();
+			for (int i = 0 ; i < instructions.Count ; i++) {
+				Instruction instruction = instructions[i];
+				if (instruction.OpCode.Code == Code.Newobj && instruction.Operand is MethodDefinition invokedMethod) {
```

Comment:
I would double check this. I think you want this to be the more general `MethodReference` instead of `MethodDefinition`. I think this code will work because it will always be within the module but that doesn't mean it couldn't be a reference.